pontoon/sync/changeset.py

```diff
@@ -234,24 +238,27 @@ def update_entity_translations_from_vcs(
                 ))
 
         # Any existing translations that were not approved get unapproved.
-        if old_translations is None:
-            old_translations = db_translations.filter(approved_date__lte=self.now)
-
         for translation in old_translations:
             if translation not in approved_translations:
+                translation = (
+                    match_attr(self.translations_to_update, pk=translation.pk) or translation
+                )
                 translation.approved = False
                 translation.approved_user = None
                 translation.approved_date = None
 
-                if translation.is_dirty():
+                if translation not in self.translations_to_update and translation.is_dirty():
                     self.translations_to_update.append(translation)
 
         # Any existing translations that are no longer fuzzy get unfuzzied.
         for translation in db_translations:
             if translation not in fuzzy_translations:
+                translation = (
```

Comment:
I think that line is worth a comment what's going in here (bugs with references to objects are tricky to find).